philander/gasgauge.py

```diff
@@ -209,19 +208,8 @@ def rateSOC( self, soc ):
         :return: The next-lower battery level mnemonic.
         :rtype: battery.Level
         """
-        ret = BatLevel.medium
-        if soc >= BatLevel.full:
-            ret = BatLevel.full
-        elif soc >= BatLevel.good:
-            ret = BatLevel.good
-        elif soc >= BatLevel.medium:
-            ret = BatLevel.medium
-        elif soc >= BatLevel.low:
-            ret = BatLevel.low
-        elif soc >= BatLevel.empty:
-            ret = BatLevel.empty
-        else:
-            ret = BatLevel.deepDischarge
+        # TODO: this function is now redundant. Should it be removed and BatLevel.from_percentage be called directly?

```

Comment:
Right. rateSOC() shall be removed from the Gas Gauge interface.
Either do it right now, as we still are in an early stage of the implementation. Or file an issue to have it done, later, please.